src/snowflake/cli/api/sql_execution.py

```diff
@@ -41,12 +41,22 @@
 
 class SqlExecutionMixin:
     def __init__(self):
-        pass
+        self._snowpark_session = None
 
     @property
     def _conn(self):
         return cli_context.connection
 
+    @property
+    def snowpark_session(self):
+        if not self._snowpark_session:
+            from snowflake.snowpark.session import Session
+
+            self._snowpark_session = Session.builder.configs(
+                {"connection": self._conn}
+            ).create()
+        return self._snowpark_session
```

Comment:
Why you open second connection?

Reply:
It's the same connection, but wrapped around Snowpark util